### PR TITLE
[UNIONVMS-4619] logbook generation for tripId

### PR DESCRIPTION
**Pull request by nomikosi**

### CODE REVIEW

File: service/src/main/java/eu/europa/ec/fisheries/ers/service/bean/JasperReportServiceBean.java [hunk 1]
```diff
@@ -0,0 +1,971 @@
...
+    private MdrModuleService mdrModuleServiceBean;
+    private HashMap<String,String> mdrGearCharacteristic;
+
+    public void generateLogBookReport(String tripId, List<FaReportDocumentEntity> faReportDocumentEntities, OutputStream destination) throws ServiceException {
+        this.mdrGearCharacteristic = getGearCharacteristicFromMDR();
+        LogbookModel logbookModel = getJasperReportData(faReportDocumentEntities, tripId);
+
+
+        try {
+            InputStream input = getClass().getClassLoader().getResourceAsStream("logbook.jrxml");
```
Comment (nikospara) [inline service/src/main/java/eu/europa/ec/fisheries/ers/service/bean/JasperReportServiceBean.java]: Unless you are absolutely certain that some component down the line is actually closing this stream, make sure it is closed:

    try (InputStream input = getClass().getClassLoader().getResourceAsStream("logbook.jrxml")) {
        JasperDesign jasperDesign = JRXmlLoader.load(input);
        ...

File: service/src/main/java/eu/europa/ec/fisheries/ers/service/bean/JasperReportServiceBean.java [hunk 2]
```diff
@@ -0,0 +1,971 @@
...
+        if(faCatchTypeEnum.equals(FaCatchTypeEnum.ONBOARD)
+                || faCatchTypeEnum.equals(FaCatchTypeEnum.KEPT_IN_NET)){
+            return CATCHES;
+        }
+
+        if(fishingActivityTypeEnum.equals(FishingActivityTypeEnum.RELOCATION) && faCatchTypeEnum.equals(FaCatchTypeEnum.LOADED) ){
+                return CATCHES;
+        }
+
+        if(fishingActivityTypeEnum.equals(FishingActivityTypeEnum.JOINT_FISHING_OPERATION) && faCatchTypeEnum.equals(FaCatchTypeEnum.UNLOADED)){
```
Comment (nikospara) [inline service/src/main/java/eu/europa/ec/fisheries/ers/service/bean/JasperReportServiceBean.java]: If `fishingActivityTypeEnum == JOINT_FISHING_OPERATION`, then you have returned `CATCHES` from line 826. Is there a logical error here (this is from IntelliJ's inspections).

File: service/src/main/java/eu/europa/ec/fisheries/ers/service/bean/JasperReportServiceBean.java [hunk 3]
```diff
@@ -0,0 +1,971 @@
...
+        }
+    }
+
+     private String formatCoordinates(FluxLocationEntity fluxLocationEntity){
+        if(fluxLocationEntity == null || fluxLocationEntity.getLongitude() == null || fluxLocationEntity.getLatitude() == null){
+            return null;
+        }
+        DecimalFormat df = new DecimalFormat("###.######");
+        df.setRoundingMode(RoundingMode.DOWN);
+        return String.valueOf(df.format(fluxLocationEntity.getLongitude()))+"/"+String.valueOf(df.format(fluxLocationEntity.getLatitude()));
```
Comment (nikospara) [inline service/src/main/java/eu/europa/ec/fisheries/ers/service/bean/JasperReportServiceBean.java]: No need for `String.valueOf()` (from IntelliJ's inspections).

File: service/src/main/java/eu/europa/ec/fisheries/ers/fa/utils/FishingActivityTypeEnum.java [hunk 4]
```diff
@@ -17,5 +17,6 @@
  */
 public enum FishingActivityTypeEnum {
     DEPARTURE, ARRIVAL, AREA_ENTRY, AREA_EXIT, FISHING_OPERATION, LANDING, TRANSHIPMENT, RELOCATION,
-    GEAR_SHOT, GEAR_RETRIEVAL, START_FISHING, JOINED_FISHING_OPERATION, START_ACTIVITY, DISCARD;
+    GEAR_SHOT, GEAR_RETRIEVAL, START_FISHING, JOINED_FISHING_OPERATION, START_ACTIVITY, DISCARD,
```
Comment (nikospara) [inline service/src/main/java/eu/europa/ec/fisheries/ers/fa/utils/FishingActivityTypeEnum.java]: Discuss with Leonidas the `JOINT`/`JOINED_FISHING_OPERATION`!

(It seems the correct solution is to drop the `JOINED_FISHING_OPERATION`, replacing it with `JOINT_FISHING_OPERATION`. Make sure the implementation of the logbook works!)

File: service/src/main/java/eu/europa/ec/fisheries/ers/service/dto/GearMapperModel.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+@Setter
+public class GearMapperModel {
+
+    private int identifier;
+    private String gearCode;
+    private String meshCode;
+    private Set<String> dimensions;
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (nikospara) [inline service/src/main/java/eu/europa/ec/fisheries/ers/service/dto/GearMapperModel.java]: Why aren't you using Lombok for `hashCode`/`equals`?

**Review by nikospara — CHANGES_REQUESTED** [review]

**Review by Georgemanif — DISMISSED** [review]

**Review by nikospara — APPROVED** [review]

**Pull request merged**